src/algorithms/mts.jl

```diff
@@ -0,0 +1,354 @@
+                end
+                if _xk_val >= xk_val
+                    update_xk = false
+                else
+                    grade += BONUS2
+                    improve[k] = true
+                end
+            else
+                grade += BONUS2
+                improve[k] = true
+            end
+        end
+        if update_xk
+            x[k] = _xk
+        end
+    end
+    return grade
+end
+
+function _localsearch3(workspace, k)
+    @unpack model, options = workspace
+    @unpack x = workspace
+    @unpack BONUS1, BONUS2 = options 
+    @unpack a_min, a_max, b_min, b_max, c_min, c_max = options
+    grade = 0
+    _xk = copy(x[k])
+    update_xk = false
+    for i in 1:model.n_dim
+        # Original value
+        _xk_val = evaluate!(model, _xk)
+        update_xk = true
+        # Heuristic search
+        _xk_x1, _xk_y1, _xk_x2  = copy(_xk), copy(_xk), copy(_xk)
+        _xk_x1[i] += 0.1
+        _xk_y1[i] -= 0.1
+        _xk_x2[i] += 0.2
+        _xk_x1_val, _xk_y1_val, _xk_x2_val = evaluate!(model, _xk_x1), evaluate!(model, _xk_y1), evaluate!(model, _xk_x2)
+        if _xk_x1_val < workspace.optimal_val
+            grade += BONUS1
+            workspace.optimal_x, workspace.optimal_ind, workspace.optimal_val = copy(_xk), k, _xk_val
+        end
+        if _xk_y1_val < workspace.optimal_val
+            grade += BONUS1
+            workspace.optimal_x, workspace.optimal_ind, workspace.optimal_val = copy(_xk), k, _xk_val
+        end
+        if _xk_x2_val < workspace.optimal_val
+            grade += BONUS1
+            workspace.optimal_x, workspace.optimal_ind, workspace.optimal_val = copy(_xk), k, _xk_val
+        end
+        D1, D2, D3 = _xk_val - _xk_x1_val, _xk_val - _xk_y1_val, _xk_val - _xk_x2_val
+        grade += ((D1>0) + (D2>0) + (D3>0)) * BONUS2
+        if update_xk == false
+            update_xk = (D1>0) || (D2>0) || (D3>0)
+        end
+        a = (rand() * (a_max - a_min)) + a_min
+        b = (rand() * (b_max - b_min)) + b_min
+        c = (rand() * (c_max - c_max)) + c_min
+        _xk[i] += a*(D1 - D2) + b*(D3 - 2*D1) + c
+    end
+    if update_xk
+        # Update x[k]
+        x[k] = _xk
+    end
+    return grade
+end
+
+# Optimization using localsearch1
+function localsearch1(workspace::MTSWorkspace)
+    M = workspace.options.M
+    for i in 1:M
+        _localsearch1(workspace, i)
+    end
+end
+
+# Optimization using localsearch2
+function localsearch2(workspace::MTSWorkspace)
+    M = workspace.options.M
+    for i in 1:M
+        _localsearch2(workspace, i)
+    end
+end
+
+# Optimization using localsearch3
+function localsearch3(workspace::MTSWorkspace)
+    M = workspace.options.M
+    for i in 1:M
+        _localsearch3(workspace, i)
+    end
+end
+
+const LOCAL_SEARCH_METHODS = [_localsearch1, _localsearch2, _localsearch3]
+const N_METHODS = length(LOCAL_SEARCH_METHODS)
```

Comment:
We should remove this.

Reply:
Any better practice for how I use them in `function MTS`?